### PR TITLE
Introduced 'brainbuilder' package

### PR DESCRIPTION
**Pull request by arsenius7**
* and its dependencies:
   * py-bluepy
   * py-bluepy-configfile
   * py-neurom
   * py-pylru
   * py-transforms3d

 * new versions of some Python packages:
   * py-chardet
   * py-click
   * py-plotly
   * py-pynrrd
   * py-pytables
   * py-requests
   * py-urllib3

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-requests/package.py [hunk 1]
```diff
@@ -23,13 +23,19 @@ class PyRequests(PythonPackage):
...
 
+    version('2.21.0', '1bcd0e0977c3f8db1848ba0e2b7ab904')
     version('2.14.2', '4c3c169ed67466088a2a6947784fe444')
     version('2.13.0', '921ec6b48f2ddafc8bb6160957baf444')
     version('2.11.1', 'ad5f9c47b5c5dfdb28363ad7546b0763')
     version('2.3.0',  '7449ffdc8ec9ac37bbcd286003c80f00')
 
     depends_on('py-setuptools', type='build')
 
+    depends_on('py-certifi@2017.4.17:', type='run')
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-requests/package.py:34]: This version does not seem to exist?
```
❯ spack info py-certifi
PythonPackage:   py-certifi

Description:
    Certifi: A carefully curated collection of Root Certificates for
    validating the trustworthiness of SSL certificates while verifying the
    identity of TLS hosts.

Homepage: http://certifi.io/

Tags:
    None

Preferred version:
    2017.1.23     https://pypi.io/packages/source/c/certifi/certifi-2017.1.23.tar.gz

Safe versions:
    2017.1.23     https://pypi.io/packages/source/c/certifi/certifi-2017.1.23.tar.gz
    2016.02.28    https://pypi.io/packages/source/c/certifi/certifi-2016.02.28.tar.gz
```

**Review by matz-e — CHANGES_REQUESTED** [review]
Looks pretty good to me, see the one version thing. Are you planning to get any of the broader package changes submitted upstream?

File: var/spack/repos/builtin/packages/py-requests/package.py (hunk 1 above)
Comment (arsenius7) [reply var/spack/repos/builtin/packages/py-requests/package.py:34]: Good catch; I guess I clicked through Spack suggestion to fetch newer `py-certifi` without checksum when testing the thing locally.

**Comment by arsenius7** [comment]
> Are you planning to get any of the broader package changes submitted upstream?

I haven't thought of version bumps as something deserving a PR to upstream.
Do we or Spack maintainers have any policy for that?

**Comment by matz-e** [comment]
> > Are you planning to get any of the broader package changes submitted upstream?
> 
> I haven't thought of version bumps as something deserving a PR to upstream.
> Do we or Spack maintainers have any policy for that?

I've pushed modifications to software not exclusive to our Spack upstream so far, to simplify rebases in the future. Not an official policy, but I think good practice. @pramodk more official opinions?

**Review by matz-e — APPROVED** [review]

**Comment by arsenius7** [comment]
Merging the change as it is for the moment.
Please let me know if we agree on the practice of pushing 3rd-party packages version bumps to upstream.

**Comment by pramodk** [comment]
>I've pushed modifications to software not exclusive to our Spack upstream so far, to simplify rebases > in the future. Not an official policy, but I think good practice. @pramodk more official opinions?

What I can think of : 

- if we add any new package (external to bbp), we should upstream
- if there is any bug fix or changes in existing spack package, we should upstream
- if we add any new bbp open sourced software package, we can/should upstream
- if we add new version, you can upstream (but we are often behind upstream, so its ok if we skip this)

For the first three, it could take few days/a week to merge to upstream (with review changes). In this case we have to do some work during rebase (as we merge to fork fast) but its not much work. So its fine.

If we don't upstream some of our PRs immediately, we can do that at the time of rebase/sync with upstream.

**Comment by pramodk** [comment]
Just minor comment : with the new PRs, I think we should also start adding dependencies to previous stages. For example, if our package (from application stage) needs some extra python package, we should add that in `deploy/packages/python-packages.yaml` as part of the same PR. This way, we don't build too many things downstream (or when we re-install from source locally).

**Pull request merged**